src/storage.rs

```diff
@@ -47,45 +32,81 @@ fn now() -> u64 {
 }
 
 pub struct Storage {
-    heap: BinaryHeap<Item>,
+    connection: Connection,
     lifetime: u64,
 }
 
 impl Storage {
-    pub fn new(lifetime: u64) -> Storage {
+    pub fn new(path: &String, lifetime: u64) -> Storage {
+        debug!("Looking for storage at `{}`", path);
+        let connection = Connection::open_with_flags(
+            Path::new(path),
+            OpenFlags::SQLITE_OPEN_READ_WRITE | OpenFlags::SQLITE_OPEN_CREATE,
+        ).unwrap();
+        connection
+            .execute(
+                "CREATE TABLE IF NOT EXISTS message (
+                    id                  INTEGER PRIMARY KEY,
+                    telegram_id         INTEGER NOT NULL,
+                    chat_telegram_id    INTEGER NOT NULL,
+                    date                TEXT NOT NULL
+                )",
+                &[],
+            )
+            .expect("Troubles during table creation");
         Storage {
-            heap: BinaryHeap::new(),
+            connection,
             lifetime,
         }
     }
 
-    pub fn add(&mut self, message: Message) {
-        self.heap.push(message.into())
+    pub fn add(&self, message: TelegramMessage) {
+        let message: Message = message.into();
+        debug!("Inserting message {:?}", message);
+        let telegram_id: i64 = message.telegram_id.into();
+        let chat_telegram_id: i64 = message.chat_telegram_id.into();
+        self.connection
+            .execute(
+                "INSERT INTO message (telegram_id, chat_telegram_id, date)
+                    VALUES (?1, ?2, ?3)",
+                &[&telegram_id, &chat_telegram_id, &message.date],
+            )
+            .expect("Troubles during message insertion");
     }
 
-    pub fn clean(&mut self) -> OutdateIter {
-        OutdateIter {
-            heap: &mut self.heap,
-            date: now() - self.lifetime,
-        }
+    pub fn clean(&self) -> Vec<Message> {
+        let threshold_date = self.threshold_date();
+        debug!("Looking for messages before {:?} to delete", threshold_date);
+        let mut statement = self
+            .connection
+            .prepare(
+                "SELECT id, telegram_id, chat_telegram_id, date
+                FROM message
+                WHERE date < ?1",
+            )
+            .unwrap();
+        let messages_iterator = statement
```

Comment:
The hardest problem for me was here. It was really hard to understand the need in this variable. The problem is that 

```rs
statement
    .query
    ...
    .filter
    ...
    .collect()
```

doesn't take ownership of iterator as far as I can see.